plugins/services/src/js/reducers/serviceForm/JSONReducers/EnvironmentVariables.js

```diff
@@ -63,20 +63,26 @@ module.exports = {
       return [];
     }
 
-    return Object.keys(state.env).reduce(function(memo, key, index) {
-      /**
+    return Object.keys(state.env)
+      .filter(function(key) {
+        return typeof state.env[key] === "string";
```

Comment:
I prefer a combination of a higher order functions instead of nested conditionals.